### PR TITLE
Create daily pool metrics spell

### PR DESCRIPTION
**Pull request by viniabussafi**
Me and @mendesfabio have been discussing this idea for a while.
Basically, the end-goal is to make DEX data more transparent, with a few of the most common metrics (swap volume, TVL and fees collected), grouped by day, project and pool, following as closely as possible the same naming pattern used on other spells on the dex sector. 
For now, the sector spell would only include balancer, but hopefully it would encourage other analysts on a combined effort to add data for other DEXs, as especially TVL and fees are not so easy to find on the spellbook as of right now. 
The dream would be to allow a dashboard such as [DEX Metrics](https://dune.com/hagaetc/dex-metrics) to be easily updated with more metrics.
Any input on refining this would be extremely valuable!

### CODE REVIEW

File: models/balancer/balancer_pools_metrics_daily.sql [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+        blockchain,
+        version,
+        pool_address AS project_contract_address,
+        pool_type,
+        pool_symbol,
+        sum(pool_liquidity_usd) AS tvl_usd,
+        sum(pool_liquidity_eth) AS tvl_eth
+    FROM {{ ref('balancer_liquidity') }}
+    {% if is_incremental() %}
+    WHERE day >= date_trunc('day', now() - interval '7' day)
```
Comment (Hosuke) [inline models/balancer/balancer_pools_metrics_daily.sql:44]: We may use `incremental_predicate` for incremental filters
```suggestion
    WHERE {{incremental_predicate('day')}}
```

File: models/balancer/balancer_pools_metrics_daily.sql [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    alias = 'pools_metrics_daily',
+    materialized = 'incremental',
+    file_format = 'delta',
+    incremental_strategy = 'merge',
+    unique_key = ['block_date', 'blockchain', 'project', 'version', 'project_contract_address'],
+    incremental_predicates = ['DBT_INTERNAL_DEST.block_date >= date_trunc(\'day\', now() - interval \'7\' day)'],
+    post_hook='{{ expose_spells(\'["arbitrum", "avalanche_c", "base", "ethereum", "gnosis", "optimism", "polygon", "zkevm"]\',
+                            "project",
+                            "balancer",
+                            \'["viniabussafi", "metacrypto"]\') }}'
```
Comment (Hosuke) [inline models/balancer/balancer_pools_metrics_daily.sql:12]:
```suggestion
    post_hook='{{ expose_spells(blockchains = \'["arbitrum", "avalanche_c", "base", "ethereum", "gnosis", "optimism", "polygon", "zkevm"]\',
                            spell_type = "project",
                            spell_name = "balancer",
                            contributors = \'["viniabussafi", "metacrypto"]\') }}'
```

File: models/balancer/balancer_pools_metrics_daily.sql [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+trades AS(
+    SELECT 
+        block_date,
+        version,
+        blockchain,
+        project_contract_address,
+        sum(amount_usd) AS swap_amount_usd
+    FROM {{ ref('balancer_trades') }}
+    {% if is_incremental() %}
+    WHERE block_date >= date_trunc('day', now() - interval '7' day)
```
Comment (Hosuke) [inline models/balancer/balancer_pools_metrics_daily.sql:27]: Same as above.

File: models/balancer/balancer_pools_metrics_daily.sql [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+fees AS(
+    SELECT
+        day,
+        version,
+        blockchain,
+        pool_address,
+        sum(protocol_fee_collected_usd) AS fee_amount_usd
+    FROM {{ ref('balancer_protocol_fee') }}
+    {% if is_incremental() %}
+    WHERE day >= date_trunc('day', now() - interval '7' day)
```
Comment (Hosuke) [inline models/balancer/balancer_pools_metrics_daily.sql:58]: Same as above.

File: models/dex/dex_pools_metrics_daily.sql [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+        alias = 'pools_metrics_daily',
+        materialized = 'incremental',
+        file_format = 'delta',
+        incremental_strategy = 'merge',
+        unique_key = ['block_date', 'blockchain', 'project', 'version', 'project_contract_address'],
+        incremental_predicates = ['DBT_INTERNAL_DEST.block_date >= date_trunc(\'day\', now() - interval \'7\' day)'],
+        post_hook='{{ expose_spells(\'["arbitrum", "avalanche_c", "base", "ethereum", "gnosis", "optimism", "polygon", "zkevm"]\',
+                                "sector",
+                                "dex",
+                                \'["viniabussafi", "metacrypto"]\') }}'
```
Comment (Hosuke) [inline models/dex/dex_pools_metrics_daily.sql:12]:
```suggestion
        post_hook='{{ expose_spells(blockchains = \'["arbitrum", "avalanche_c", "base", "ethereum", "gnosis", "optimism", "polygon", "zkevm"]\',
                                spell_type = "sector",
                                spell_name = "dex",
                                contributors = \'["viniabussafi", "metacrypto"]\') }}'
```

File: models/dex/dex_pools_metrics_daily.sql [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+        project_contract_address,
+        pool_symbol,
+        pool_type,
+        swap_amount_usd,
+        tvl_usd,
+        tvl_eth,
+        fee_amount_usd
+    FROM {{ dex_model }}
+    {% if is_incremental() %}
+    WHERE block_date >= date_trunc('day', now() - interval '7' day)
```
Comment (Hosuke) [inline models/dex/dex_pools_metrics_daily.sql:38]: Same as above.

**Review by Hosuke — CHANGES_REQUESTED** [review]
I left some comments mostly about the new syntax.

**Comment by viniabussafi** [comment]
thanks for the review, @Hosuke! fixed what you mentioned

File: models/dex/dex_pools_metrics_daily.sql [hunk 7]
```diff
@@ -0,0 +1,44 @@
+{{ config(
+        
+        alias = 'pools_metrics_daily',
```
Comment (Hosuke) [inline models/dex/dex_pools_metrics_daily.sql:3]:
```suggestion
        schema = 'dex'
        alias = 'pools_metrics_daily',
```

**Review by Hosuke — APPROVED** [review]
LGTM.✅
Thank you @viniabussafi 

[Sample balancer_pools_metrics_daily](https://dune.com/queries/3710747)

Probably a good idea for `dex` sector @jeff-dude

File: models/dex/dex_pools_metrics_daily.sql [hunk 8]
```diff
@@ -0,0 +1,44 @@
+{{ config(
+        schema = 'dex'
```
Comment (Hosuke) [inline models/dex/dex_pools_metrics_daily.sql:2]:
```suggestion
        schema = 'dex',
```

File: models/balancer/balancer_schema.yml [hunk 9]
```diff
@@ -405,4 +405,35 @@ models:
...
+    config:
+      tags: ['arbitrum', 'avalanche_c', 'base', 'ethereum', 'gnosis', 'optimism', 'polygon', 'zkevm', 'balancer', 'pool', 'stats', 'volume', 'tvl', 'fee']
+    description: >
+      This spell aggregates data from the trades, liquidity and protocol fees spells, by day and pool, while also displaying some basic information about the pool
+    tests:
+      - dbt_utils.unique_combination_of_columns:
+          combination_of_columns:
+            - block_date
+            - blockchain
+            - project_contract_address
```
Comment (jeff-dude) [inline models/balancer/balancer_schema.yml:423]: ideally we match this test with columns in unique_keys of the model. can you update one of them to reflect the true keys?

File: models/dex/dex_schema.yml [hunk 10]
```diff
@@ -612,4 +612,36 @@ models:
...
+    config:
+      tags: ['dex', 'pools', 'volume', 'fee', 'tvl']
+    description: >
+      This spell aggregates data from the trades, liquidity and protocol fees spells, by day and pool, while also displaying some basic information about the pool
+    tests:
+      - dbt_utils.unique_combination_of_columns:
+          combination_of_columns:
+            - block_date
+            - blockchain
+            - project_contract_address
```
Comment (jeff-dude) [inline models/dex/dex_schema.yml:631]: same idea here

File: models/dex/dex_pools_metrics_daily.sql
Comment (jeff-dude) [inline models/dex/dex_pools_metrics_daily.sql:1]: you should likely move the `dex` spells in this PR to the new `_sector/dex/..` path, as we would like to deprecate this old path once we clean everything up 🙏

**Review by jeff-dude — CHANGES_REQUESTED** [review]
left some minor comments, but i agree in general with this being a valuable downstream cross-chain/sector-wide spell. in the future, when we complete some of the improvements to scale spellbook, we would love to help internally build these out with you and others. for now, we will need to rely on community analysts to help us build it out 🫡

File: models/balancer/balancer_schema.yml (hunk 9 above)
Comment (viniabussafi) [reply models/balancer/balancer_schema.yml]: yep, sorry for the oversight

**Comment by viniabussafi** [comment]
should be good to go now, @jeff-dude! thanks for the review

File: models/_sector/dex/pools_metrics_daily/_schema.yml
Comment (jeff-dude) [inline models/_sector/dex/pools_metrics_daily/_schema.yml:1]: going to be a bit nitpicky here :)
let's just use the existing `_sector/dex/pools/` directory

**Review by jeff-dude — CHANGES_REQUESTED** [review]

File: models/_sector/dex/pools_metrics_daily/_schema.yml
Comment (viniabussafi) [reply models/_sector/dex/pools_metrics_daily/_schema.yml:1]: done!

**Review by jeff-dude — APPROVED** [review]
sounds good!

**Pull request merged**